### PR TITLE
OP-400  increase coverage in priceslist and pricesothers; format Manager file 

### PR DESCRIPTION
**Pull request by dbmalkovsky**
Coverage before:
image

Coverage after:
image

**Note**: there are no tests for `org.isf.priceslist.model.List` as there are no occurrences where the class is used in any of the OH repos.  See [OP-399](https://openhospital.atlassian.net/browse/OP-399).

### CODE REVIEW

**Comment by dbmalkovsky** [comment]
The note about the `List` class no longer applies as @mwithi has pushed the changes with [OP-399](https://openhospital.atlassian.net/browse/OP-399) to remove the class.

File: src/test/java/org/isf/priceslist/test/Tests.java [hunk 1]
```diff
@@ -212,6 +220,249 @@ public void testIoCopyListSteps() throws Exception {
...
+	@Test
+	public void testPriceEquals() throws Exception {
+		PriceList priceList = testPriceList.setup(true);
+		Price price = new Price(priceList, "TG", "TestItem", "TestDescription", 10.10);
+
+		assertThat(price.equals(price)).isTrue();
+		assertThat(price).isNotEqualTo(null);
+		assertThat(price).isNotEqualTo("someString");
+
+		PriceList priceList2 = testPriceList.setup(true);
```
Comment (emecas) [inline src/test/java/org/isf/priceslist/test/Tests.java:397]: `priceList2 `is not used

File: src/test/java/org/isf/pricesothers/test/Tests.java [hunk 2]
```diff
@@ -127,6 +134,109 @@ public void testIoDeletePricesOthers() throws Exception {
...
+	@Test
+	public void testPricesOthersHashCode() throws Exception {
+		PricesOthers pricesOthers = testPricesOthers.setup(true);
+		pricesOthers.setId(1);
+		int hashCode = pricesOthers.hashCode();
+		assertThat(hashCode).isEqualTo(23 * 133 + 1);
+		// check computed value
+		assertThat(pricesOthers.hashCode()).isEqualTo(hashCode);
+	}
+
```
Comment (emecas) [inline src/test/java/org/isf/pricesothers/test/Tests.java:239]: testing is doing what it supposes to do, but why this `hashCode` method is only involving 1 single attribute from `PricesOthers` class?, there are 6 more attributes: `code`, `desc`, `opdInclude`, `ipdInclude`, `discharge`, and  `undefined`

cc @mwithi

**Review by emecas — COMMENTED** [review]
Thanks @dbmalkovsky 

Coverage is much better now and all tests are passing, just a couple observations

```
[17/Jan/2021 18:01:57] [:] INFO - Closing JPA EntityManagerFactory for persistence unit 'default'

Results :

Tests run: 782, Failures: 0, Errors: 0, Skipped: 1

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:06 min
[INFO] Finished at: 2021-01-17T18:01:57-05:00
[INFO] ------------------------------------------------------------------------
```

File: src/test/java/org/isf/priceslist/test/Tests.java (hunk 1 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/priceslist/test/Tests.java:397]: Good catch; changed the test.

File: src/test/java/org/isf/pricesothers/test/Tests.java (hunk 2 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/pricesothers/test/Tests.java:239]: @emecas  I just test 'em 😄 and rarely do I question the implementations (except when they are *obviously* very wrong).

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java [hunk 3]
```diff
@@ -42,97 +42,100 @@
...
 	private PricesListIoOperations ioOperations;
-	
+
 	/**
-	 * Return the list of {@link PriceList}s in the DB
-	 * @return the list of {@link PriceList}s
-	 * @throws OHServiceException 
+	 * Return the list of {@link List}s in the DB
+	 *
+	 * @return the list of {@link List}s
```
Comment (mwithi) [inline src/main/java/org/isf/priceslist/manager/PriceListManager.java:49]: This was fixed in https://github.com/informatici/openhospital-core/commit/947756e901c67d3b47ab2374857e28b015ef3644

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java [hunk 4]
```diff
@@ -42,97 +42,100 @@
...
-        return ioOperations.getPrices();
+		return ioOperations.getPrices();
 	}
 
 	/**
-	 * Updates all {@link Price}s in the specified {@link PriceList}
-	 * @param list - the {@link PriceList}
+	 * Updates all {@link Price}s in the specified {@link List}
+	 *
+	 * @param list - the {@link List}
```
Comment (mwithi) [inline src/main/java/org/isf/priceslist/manager/PriceListManager.java:69]: This also

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java [hunk 5]
```diff
@@ -42,97 +42,100 @@
...
+		return ioOperations.updatePrices(list, prices);
 	}
 
 	/**
-	 * Insert a new {@link PriceList} in the DB
-	 * 
-	 * @param list - the {@link PriceList}
+	 * Insert a new {@link List} in the DB
+	 *
+	 * @param list - the {@link List}
```
Comment (mwithi) [inline src/main/java/org/isf/priceslist/manager/PriceListManager.java:81]: this too

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java [hunk 6]
```diff
@@ -42,97 +42,100 @@
...
+		return ioOperations.newList(list);
 	}
 
 	/**
-	 * Update a {@link PriceList} in the DB
-	 * 
-	 * @param updateList - the {@link PriceList} to update
+	 * Update a {@link List} in the DB
+	 *
+	 * @param updateList - the {@link List} to update
```
Comment (mwithi) [inline src/main/java/org/isf/priceslist/manager/PriceListManager.java:93]: like above

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java [hunk 7]
```diff
@@ -42,97 +42,100 @@
...
+		return ioOperations.updateList(updateList);
 	}
 
 	/**
-	 * Delete a {@link PriceList} in the DB
-	 * 
-	 * @param deleteList - the {@link PriceList} to delete
+	 * Delete a {@link List} in the DB
+	 *
+	 * @param deleteList - the {@link List} to delete
```
Comment (mwithi) [inline src/main/java/org/isf/priceslist/manager/PriceListManager.java:105]: same

**Review by mwithi — CHANGES_REQUESTED** [review]
It seems the rebase https://github.com/informatici/openhospital-core/pull/285/commits/f31746e0073ad009077675e8a7344b14a38fba41 was not rebasing at all since some changes seem lost.

File: src/test/java/org/isf/pricesothers/test/Tests.java (hunk 2 above)
Comment (mwithi) [reply src/test/java/org/isf/pricesothers/test/Tests.java:239]: Me too I didn't question that implementation at its time, but the same code is quite in every model's class at the first porting to JPA. I think we can fly over it for this PR and discuss it in another place.

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java (hunk 4 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/priceslist/manager/PriceListManager.java]: I believe I've resolved (merged) the changes from OP-399

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java (hunk 5 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/priceslist/manager/PriceListManager.java]: I believe I've resolved (merged) the changes from OP-399

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java (hunk 6 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/priceslist/manager/PriceListManager.java]: I believe I've resolved (merged) the changes from OP-399

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java (hunk 7 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/priceslist/manager/PriceListManager.java]: I believe I've resolved (merged) the changes from OP-399

File: src/main/java/org/isf/priceslist/manager/PriceListManager.java (hunk 3 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/priceslist/manager/PriceListManager.java]: I believe I've resolved (merged) the changes from OP-399

**Review by mwithi — COMMENTED** [review]
As a standalone component `core` looks good, with tests sucessful, but the PR seems causing compilation errors in `gui` and `api` ones, @dbmalkovsky could you check please?
```
Results :

Tests run: 970, Failures: 0, Errors: 0, Skipped: 1
```

**Comment by dbmalkovsky** [comment]
@mwithi I'll take a look but doesn't it seem strange that adding tests and formatting the manger files would cause compile errors?
Can you give me some indication of what you are seeing that is failing?

**Comment by mwithi** [comment]
> @mwithi I'll take a look but doesn't it seem strange that adding tests and formatting the manger files would cause compile errors?
> Can you give me some indication of what you are seeing that is failing?

image

**Comment by dbmalkovsky** [comment]
The problem is that https://github.com/informatici/openhospital-core/pull/295 which changed the return types for Vaccine and Disease is not applied to that branch.   It has nothing to do with the changes in OP-400.

**Pull request merged**